internal/server/sshd.go

```diff
@@ -231,6 +230,20 @@ func StartSSHServer(sshListener net.Listener, privateKey ssh.Signer, insecure bo
 			}, nil
 
 		},
+		PasswordCallback: func(conn ssh.ConnMetadata, password []byte) (*ssh.Permissions, error) {
+			params := make(map[string]string, 0)
+			res, err := Post(conf.ExternalAuthApi, params)
+			if err != nil {
+				return nil, fmt.Errorf("call external auth api %v get error", conf.ExternalAuthApi)
```

Comment:
Does this even do anything?? `params` is just an empty map? And if the HTTP request doesn't give an error, it just allows auth...?

Reply:
Ah good point, this would not working as an authentication method, I was a little bit focused on the bits that were removing public key authentication which is explicitly against what I want to do.

Reply:
Sorry, this was an oversight, forgot to wear the username and password, I already have a new commit, but not sure how to update the pr, I'm still working on it

Reply:
What is your intention for the response to the POST request?

You just take the `res`, and assign that as the `type` in the permissions extensions? If the API returns "client", it will allow a client binary to connect with SSH. If it returns "user", it will allow a human operator to connect. 

How would the server respond to an authentication failure? I don't see any handling for that? If the server responds with something other than "client" or "user", then even though `acceptConn` would close the connection, the user is still allowed to authenticate with completely incorrect credentials...

Frankly, I find it quite concerning that the first version of this code didn't even send the credentials, and the second version doesn't even handle the response properly, allowing ANY username or password to authenticate...

Reply:
Only the correct user can pass the authorization and return the clientType, otherwise it will return non-200, and after the request is responded to, all non-200 will return an error

Reply:
> all non-200 will return an error

This is the opposite of the truth

> A non-2xx status code doesn't cause an error. 

https://pkg.go.dev/net/http#Client.Do

Reply:
i mean we can implement it in api itself,any way i have a new commit is meet your wanna